lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1441,43 +1447,39 @@ void DifferentiableActivityInfo::analyze(DominanceInfo *di,
           for (auto result : inst.getResults())
             if (isUseful(result, i))
               for (auto &op : inst.getAllOperands())
-                setUsefulIfDifferentiable(op.get(), i);
+                setUseful(op.get(), i);
         }
       }
     }
     postDomOrder.pushChildren(block);
   }
 }
 
-bool DifferentiableActivityInfo::setVariedIfDifferentiable(
+bool DifferentiableActivityInfo::setVaried(
     SILValue value, unsigned independentVariableIndex) {
-  if (!value->getType().isDifferentiable(function.getModule()))
-    return false;
   variedValueSets[independentVariableIndex].insert(value);
   return true;
 }
 
-bool DifferentiableActivityInfo::setUsefulIfDifferentiable(
+bool DifferentiableActivityInfo::setUseful(
     SILValue value, unsigned dependentVariableIndex) {
-  if (!value->getType().isDifferentiable(function.getModule()))
-    return false;
   usefulValueSets[dependentVariableIndex].insert(value);
   return true;
```

Comment:
Since this no longer checks the type (and always returns true), the return value should be eliminated.